examples/inspection/plot_linear_model_coefficient_interpretation.py

```diff
@@ -0,0 +1,274 @@
+    X, y, random_state=42
+)
+
+##############################################################################
+# Then we fit the model
+
+model.fit(X_train, y_train)
+
+##############################################################################
+# We can check the performance of the computed model using, for example, the
+# median absolute error of the model.
+
+from sklearn.metrics import median_absolute_error
+
+y_pred = model.predict(X_train)
+mae = median_absolute_error(y_train, y_pred)
+string_score = 'MAE on training set: {0:.2f} $/hour'.format(mae)
+y_pred = model.predict(X_test)
+mae = median_absolute_error(y_test, y_pred)
+
+string_score += '\nMAE on testing set: {0:.2f} $/hour'.format(mae)
+
+fig, ax = plt.subplots(figsize=(6, 6))
+sns.regplot(y_test, y_pred)
+
+plt.text(3, 20, string_score)
+
+plt.ylabel('Model predictions')
+plt.xlabel('Truths')
+plt.xlim([0, 27])
+plt.ylim([0, 27])
+
+##############################################################################
+# The model learnt is far to be a good model making accurate prediction.
+# As interpretation tools characterize model rather than the generative process
+# of the data itself, it needs to be emphasized that interpretations are
+# correct if the model is correct as well.
+#
+# Interpreting coefficients
+# .........................
+#
+# First of all, we can plot the values of the coefficients of the regressor we
+# have fitted.
+
+feature_names = (model.named_steps['columntransformer']
+                      .named_transformers_['onehotencoder']
+                      .get_feature_names(input_features=categorical_columns))
+feature_names = np.concatenate(
+    [feature_names, binary_columns, numerical_columns])
+
+coefs = pd.DataFrame(
+    model.named_steps['transformedtargetregressor'].regressor_.coef_,
+    columns=['Coefficients'], index=feature_names
+)
+coefs.plot(kind='barh', figsize=(9, 7))
+plt.axvline(x=0, color='.5')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# Soon we realize that we cannot compare different coefficients since we did
+# not scale the data before the fit, features having different value ranges.
+# For instance, the "AGE" coefficient is expressed in $/hours/leaving years
+# while the "EDUCATION" is expressed in $/hours/years of education.
+# This is evident if we compare feature standard deviations.
+
+X_train_preprocessed = pd.DataFrame(
+    model.named_steps['columntransformer'].transform(X_train),
+    columns=feature_names
+)
+X_train_preprocessed.std().plot(kind='barh', figsize=(9, 7))
+plt.title('Features std. dev.')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# We should then normalize the coefficients by the standard deviation and we
+# will be able to compare them.
+
+coefs = pd.DataFrame(
+    model.named_steps['transformedtargetregressor'].regressor_.coef_ *
```

Comment:
Should this be a division?

Reply:
If I understand correctly, this is a way to describe all the coefficients with the same unit of measurement.  the AGE coefficient is expressed in $/hours/leaving years while the EDUCATION is expressed in $/hours/years of education. Variances are expressed respectively in "leaving years" and "years of education". In order to express both coefficients in $/hours and make them comparable we need a multiplication here. 
I've changed the text a bit because suddenly I realised that I wasn't comfortable with the explanation...

Reply:
normalizing usually means dividing by something

Dividing by the std dev has the advantage of comparing unitless entities.

Perhaps @GaelVaroquaux can clarify since I believe this originally comes from you?